### PR TITLE
[RELEASE] v0.6 cuDF Release

### PR DESCRIPTION
**Pull request by mike-wendt**
### What does this mean?
Only hotfix and/or critical bug level issues should be merged into `branch-0.6` until release (merging of this PR).

### What is the purpose of this PR?
- Update documentation
- Allow testing for the new release
- Enable a means to merge `branch-0.6` into `master` for the release

### CODE REVIEW

**Comment by thomcom** [comment]
rerun tests

**Comment by raydouglass** [comment]
rerun tests

**Comment by raydouglass** [comment]
rerun tests

**Comment by raydouglass** [comment]
rerun tests

**Comment by dillon-cullinan** [comment]
rerun tests

**Comment by harrism** [comment]
Can we merge the latest branch-0.6 into this PR to get the latest PRs from that branch?

**Comment by raydouglass** [comment]
rerun tests

**Comment by raydouglass** [comment]
> Can we merge the latest branch-0.6 into this PR to get the latest PRs from that branch?

Since this PR is `branch-0.6`->`master`, any PRs merged into `branch-0.6` will automatically be added to this PR!

**Comment by harrism** [comment]
@mike-wendt @raydouglass needs a changelog.

CI is failing to find libcuda?

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,12 +163,14 @@
 
 ## New Features
 
+- PR #722 Add bzip2 decompression support to `read_csv()`
```
Comment (harrism) [inline CHANGELOG.md]: Why is the 0.6 release modifying CHANGELOG for 0.5.1?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,157 @@
+# cuDF 0.6.0 (Date TBD)
```
Comment (harrism) [inline CHANGELOG.md]: Needs a date.

**Comment by raydouglass** [comment]
rerun tests

**Pull request merged**